modules/apps/asset/asset-categories-navigation-web/src/main/java/com/liferay/asset/categories/navigation/web/internal/upgrade/v1_0_2/UpgradePortletPreferences.java

```diff
@@ -0,0 +1,145 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.asset.categories.navigation.web.internal.upgrade.v1_0_2;
+
+import com.liferay.asset.categories.navigation.constants.AssetCategoriesNavigationPortletKeys;
+import com.liferay.asset.kernel.model.AssetVocabulary;
+import com.liferay.asset.kernel.service.AssetVocabularyLocalService;
+import com.liferay.portal.kernel.util.ArrayUtil;
+import com.liferay.portal.kernel.util.GetterUtil;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.portal.kernel.util.Validator;
+import com.liferay.portlet.display.template.upgrade.BaseUpgradePortletPreferences;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.portlet.PortletPreferences;
+
+/**
+ * @author Mikel Lorza
+ */
+public class UpgradePortletPreferences extends BaseUpgradePortletPreferences {
+
+	public UpgradePortletPreferences(
+		AssetVocabularyLocalService assetVocabularyLocalService) {
+
+		_assetVocabularyLocalService = assetVocabularyLocalService;
+	}
+
+	@Override
+	protected String[] getPortletIds() {
+		return new String[] {
+			AssetCategoriesNavigationPortletKeys.ASSET_CATEGORIES_NAVIGATION +
+				"_INSTANCE_%"
+		};
+	}
+
+	@Override
+	protected void upgradePreferences(
+			long companyId, long ownerId, int ownerType, long plid,
+			String portletId, PortletPreferences portletPreferences)
+		throws Exception {
+
+		long displayStyleGroupId = GetterUtil.getLong(
+			portletPreferences.getValue("displayStyleGroupId", null));
+
+		if (displayStyleGroupId > 0) {
+			String scopeExternalReferenceCode = getGroupExternalReferenceCode(
+				displayStyleGroupId);
+
+			if (Validator.isNotNull(scopeExternalReferenceCode)) {
+				portletPreferences.reset("displayStyleGroupId");
+				portletPreferences.setValue(
+					"displayStyleGroupExternalReferenceCode",
+					scopeExternalReferenceCode);
+			}
+		}
+
+		String assetVocabularyIdsString = portletPreferences.getValue(
+			"assetVocabularyIds", null);
+
+		if (Validator.isNull(assetVocabularyIdsString)) {
+			return;
+		}
+
+		long[] assetVocabularyIds = GetterUtil.getLongValues(
+			StringUtil.split(assetVocabularyIdsString, ','));
+
+		if (ArrayUtil.isEmpty(assetVocabularyIds)) {
+			return;
+		}
+
+		Map<Long, List<String>> groupAssetVocabularyIdsMap =
+			_getGroupAssetVocabularyExternalReferenceCodesMap(
+				assetVocabularyIds);
```

Comment:
Shouldn't this map be called `groupAssetVocabularyExternalReferenceCodesMap`? Usually, we name the map based on the values, not the keys.